### ISSUE TITLE
Brace expansion with first term / command

### ISSUE THREAD

**Issue by razeetg** [fish-shell/fish-shell#5328]
Brace expansion for first term (command) is not the same as it is for the rest of the command line. For example, if I enter:
```
> echo ech{a,b}
```
It gives the expected output of 
```
echa echb
```

However if I enter 
```
> ech{o,a}
```
I get
```
fish: Illegal command name “ech{o,a}”
```
I would expect it to expand to echo echa and execute echo with argument echa

Also to note,
```
ech{o} a
```
works as expected, and outputs `a`

Also just entering `{}` or `{,}` gives an error like  `fish: The file “” is not executable by this user`, where I would expect it to be treated as an empty command, just like pressing enter

I tested on the following:
Fish version: 2.7.1
OS: Debian 9.5
Terminal Emulator: Konsole
I tried this with HOME=tempdir also to make sure its not an add on or theme.

**Comment by razeetg** [comment]
Also as an additional note, just entering `{}` or `{,}` gives the error  `fish: The file “” is not executable by this user`, where I would expect it to be treated as an empty command, just like pressing enter

**Comment by zanchey** [comment]
This is a duplicate of #154, and you'll be pleased to know it is fixed in the soon-to-be-released fish 3.0.
